### DISCUSSION TITLE
[Feature Request] Add placeholderColor.DEFAULT

### DISCUSSION THREAD

**Discussion by cwgw** [tailwindlabs/tailwindcss#6966] [Ideas]
Hello!

I'd like to propose adding a `DEFAULT` key to `theme.placeholderColor` and referencing it in preflight.

Some core plugins (such as ringColor, borderColor and dropShadow) use a `DEFAULT` value, while others do not (such as placeholderColor). I don't know how this choice gets made, but I'd like to make a pitch for adding `DEFAULT` to `placeholderColor`.

Currently in `preflight.css`, theme colors are referenced twice: [(1)](https://github.com/tailwindlabs/tailwindcss/blob/master/src/css/preflight.css#L12) to set the default border color…

```css
border-color: theme('borderColor.DEFAULT', 'currentColor'); /* 2 */
```

…and [(2)](https://github.com/tailwindlabs/tailwindcss/blob/master/src/css/preflight.css#L314) to set the default placeholder color…

```css
color: theme('colors.gray.400', #9ca3af); /* 2 */
```

This feels like an oversight.

To change the default placeholder color, I need to either redeclare it with `@base` in some stylesheet (which I might not have) or set `colors.gray.400` to my desired placeholder color (which might not be feasible). It would be great if I could set `placeholderColor.DEFAULT` in my config as I do with `borderColor.DEFAULT` and `ringColor.DEFAULT`.

What do you think?

I would be happy to provide a PR. I think we can add this without breaking changes:

```diff
// src/css/preflight.css

input::placeholder,
textarea::placeholder {
  opacity: 1; /* 1 */
-  color: theme('colors.gray.400', #9ca3af); /* 2 */
+  color: theme('placeholderColor.DEFAULT', #9ca3af); /* 2 */
}

// stubs/defaultConfig.stub.js

    padding: ({ theme }) => theme('spacing'),
-    placeholderColor: ({ theme }) => theme('colors'),
+    placeholderColor: ({ theme }) => ({
+      DEFAULT: theme('colors.gray.400', #9ca3af),
+      ...theme('colors'),
+    }),
    placeholderOpacity: ({ theme }) => theme('opacity'),
```

**Comment by 0x221A** [comment]
I agreed with your idea and ended up patching tailwindcss myself.
